### PR TITLE
✨ Source Intercom: add new substream ContactSegments

### PR DESCRIPTION
**Pull request by nisargpatel-m**
## What
adding a substream fetching the segments object individually from the contacts streams, adding an additional contact id field to the substream

feat(source-intercom): add contact_segments stream with Dockerfile setup

This commit introduces the 'contact_segments' stream to the source-intercom connector, extending its functionality. The update includes a new manifest section, JSON schema file, and the necessary Dockerfile configuration for proper environment setup.

- The Dockerfile has been created to set up a lightweight Python environment using Alpine Linux as a base image.

- A new manifest entry for the 'contact_segments' substream outlines incremental sync capabilities along with requisite transformations.

- The corresponding JSON schema for 'contact_segments' specifies its data structure and types.

This enhancement allows users to retrieve segment information associated with contacts directly from Intercom through Airbyte's integration system, improving the completeness of available data for synchronization tasks.

## How
Creating a new substream i.e. contact_segment based of the parent stream i.e. contact. This substream waits till the parent stream has been completed running and than retrieves the segment information for each of the contact. 

## Recommended reading order
1. `manifest.yaml`
2. `schema/contact_segments.json`
3. `Dockerfile`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

This pr introduces **_no_** breaking changes.

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

File: airbyte-integrations/connectors/source-intercom/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+FROM python:3.9.11-alpine3.15 as base
+
+# build and load all requirements
+FROM base as builder
```
[the remark this answers is not in the document]
Comment (nisargpatel-m) [reply airbyte-integrations/connectors/source-intercom/Dockerfile:4]: Good to know, In that case Ill take away the docker file from the PR

**Comment by marcosmarxm** [comment]
@nisargpatel-m please make sure to sign the CLA or use the same user to commit the changes.

**Comment by marcosmarxm** [comment]
Closed due inactivity @nisargpatel-m let me know if you want to return this contribution

**Pull request closed without merging**